src/NLog/Targets/MailTarget.cs

```diff
@@ -449,6 +451,26 @@ internal void ConfigureMailClient(LogEventInfo lastEvent, ISmtpClient client)
             
         }
 
+        /// <summary>
+        /// Handle <paramref name="pickupDirectoryLocation"/> if it is a virtual directory. Set PickupDirectoryLocation of <paramref name="client"/>
+        /// </summary>
+        /// <param name="client">client to set properties on</param>
+        /// <param name="pickupDirectoryLocation"></param>
+        /// <returns></returns>
+        internal static string GetPickupDirectoryLocation(ISmtpClient client, string pickupDirectoryLocation)
```

Comment:
Proposal: 
- remove 'ISmtpClient client' (I guess this is a bug)
- rename to "ConvertDirectoryLocation"
- update docs